MSliceMainView.py

```diff
@@ -0,0 +1,221 @@
+#This file in its current state only exists for reference
+#import abc
+
+class MainView(object):
+    #__metaclass__ = abc.ABCMeta
+    #TODO metaclass?
+
+    #@abc.abstractmethod
+    def update_log(self, log):
+        pass
+
+    # <These Might go>
```

Comment:
You say that "these might go". I take it that they are effectively dead-code then. 

I strongly suggest that you ONLY add methods to your MainView as you go along. In other words delete out all the methods that you are not currently using. They'll only get in your way, slow you down, and mislead your design.